### PR TITLE
Implemented Post Creation, Retrieval, and Deletion Features

### PR DESCRIPTION
**Pull request by Lilemanalu**

### CODE REVIEW

File: microservices/post-service/src/main/java/mini_twitter/post_service/service/PostService.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            }
+        } else {
+            logger.error("Post ID: {} not found or user ID: {} is not authorized to delete this post.", postId, userId);
+            return WebResponseDto.<String>builder()
+                    .errors("Post not found or you are not authorized to delete this post.")
+                    .build();
+        }
+    }
+
+    @Transactional
```
Comment (install18) [inline microservices/post-service/src/main/java/mini_twitter/post_service/service/PostService.java]: Why Fetching data need annotation @Transactional ?

File: microservices/post-service/src/main/java/mini_twitter/post_service/service/PostService.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                    .build();
+        } else {
+            logger.warn("Post not found for postId: {}", postId);
+            return WebResponseDto.<PostDetailResponseDto>builder()
+                    .errors("Post not found.")
+                    .build();
+        }
+    }
+
+    @Transactional
```
Comment (install18) [inline microservices/post-service/src/main/java/mini_twitter/post_service/service/PostService.java]: Why Fetching data need annotation @Transactional ?

Comment (Lilemanalu) [reply microservices/post-service/src/main/java/mini_twitter/post_service/service/PostService.java]: Thanks for your feedback! I initially added @Transactional to the data-fetching method thinking it was necessary. I understand now that it’s not needed for read-only operations, so I’ve removed it and added try-catch blocks for error handling instead.

**Comment by install18** [comment]
LGTM!

**Pull request merged**